[Feat] GNB-Component / Login-page / Sign-page

YongLeeCode opened the pull request:
<!-- PR 제목은 " [Feat] landing-page" 형식으로 작성 -->
<!-- PR open 후 링크를 복사해 노션에 추가 -->

## ✅ 체크 리스트

<!-- PR 승인을 받기전 놓친 것 없는지 확인해주세요. -->

- [ ] 적절한 Title 작성
- [ ] 적절한 Label 지정
- [ ] Assignee 및 Reviewer 지정
- [ ] 로컬 작동 확인
- [ ] Merge 되는 브랜치 확인

## 📝 작업 내용

로그인 및 회원가입 서버 통신된 버전입니다.
PR이 merge되면 자유롭게 로그인과 회원가입이 가능할 겁니다.
로그아웃은 가능하나, 백엔드 통신에는 문제가 있어서 주석 처리된 상태입니다.
그러나 로그아웃 버튼은 사용가능하다는 것 알아주세요.

전에 merge했던 GNB가 refresh가 계속 반복되는 버그가 있어서 해결했습니다.

YongLeeCode committed:
refactor: eslint 문제

YongLeeCode committed:
feat: json-server 데이터 수정

YongLeeCode committed:
Refactor: PR 리뷰 반영

YongLeeCode committed:
feat: zustand 설치

YongLeeCode committed:
feat: 서버 변경 및 로그아웃 기능 생성

YongLeeCode committed:
feat: zustand 폴더 생성

YongLeeCode committed:
feat: 로그아웃 버튼 활성화 및 로그인 상태 zustand로 저장

YongLeeCode committed:
feat: 토글 로그아웃 기능 활성화

YongLeeCode committed:
feat: env 파일 생성

YongLeeCode committed:
refactor: GNB conflict 수정

YongLeeCode committed:
feat: env 파일 gitignore에 추가

YongLeeCode committed:
Merge branch 'dev' into component/gnb

YongLeeCode committed:
feat: 반복 리프레쉬  수정

YongLeeCode committed:
feat: 로그인 서버 연결

YongLeeCode committed:
feat: 회원가입 서버 연결

han9898 commented on .env:
```diff
@@ -0,0 +1 @@
+NEXT_PUBLIC_API_URL ='http://13.124.98.133'
```
p1
git pull 하셨을까요? 최신 origin dev의 `.gitignore` 파일 안에 .env가 추가되어 있음에도 이 파일이 추가 됐네요!! 확인해주세요

YongLeeCode replied:
이미 푸쉬한 건 따로 지워야했네요 ㅎㅎ;;
해결했습니다.

inji0212 reviewed (approved):
소희님 p1만 해결되면 될 것 같습니다!

YongLeeCode committed:
feat: .env 파일 깃헙으로 푸쉬에서 제외

YongLeeCode committed:
feat: Toastify 생성

- 경고 혹은 성공 팝업 창을 이한 Toastify 생성
- 403 로그아웃 버그 해결, 그러나 400 에러가 여전히 발생함으로 주석 처리 중
- accessToken 변수로 가지고 오지 않고 바로 적용

YongLeeCode committed:
refactor: accessToken 파라미터 제거

YongLeeCode committed:
feat: Toastify 생성

- 경고 혹은 성공 팝업 창을 위한 Toastify 생성
- 이름 중복 확인 서버 연결

YongLeeCode committed:
feat: Toastify 생성

han9898 reviewed (approved):
좋아요~ 회원가입 페이지 pc 일 때 오른쪽 캐러셀 부분에 마우스 호버하면 색상이 바뀌어요! 나중에 변경해주세요👍👍

edhcoding reviewed (approved):
추가적인 디자인 문의는 인지님한테 하겠습니다 고생하셨어요!

YongLeeCode committed:
feat: 조건부 엑세스 토큰 담는 인터셉터 생성

YongLeeCode committed:
refactor: 조건부 인터셉터를 사용으로 인해 코드 생략

YongLeeCode committed:
feat: 3자 이상 일 때 중복 체크 시작

YongLeeCode committed:
refactor: carousel 호버 삭제
